src/options.js

```diff
@@ -490,3 +490,11 @@ export function named(things) {
 export function maybeNamed(things) {
   return isIterable(things) ? named(things) : things;
 }
+
+export function isTimeInterval(t) {
+  return isInterval(t) && typeof t === "function" && t() instanceof Date;
```

Comment:
This disagrees with how we’ve defined intervals in TypeScript, where they are objects rather than decorated functions:

https://github.com/observablehq/plot/blob/820144e4a89b07f2a212f66156b4af0dc48a250b/src/interval.d.ts#L19-L22

And this means that you can use a {floor, offset, range?} object in Plot, which works, except when we try to do this time interval test.

Also, I don’t think we actually need this restriction, because D3 doesn’t require that time intervals are functions, either. It instead checks for the range method (and then blindly invokes the floor and offset methods):

https://github.com/d3/d3-time/blob/9e8dc940f38f78d7588aad68a54a25b1f0c2d97b/src/ticks.js#L38
https://github.com/d3/d3-scale/blob/2b7db622b1a224d9ea19ff15c4cc8cbb3b25f4a4/src/time.js#L58

I added that duck testing in #572, but I think it would be better to replace that duck testing with something consistent with our new TypeScript declarations and with D3. Probably we should invoke the floor method and see if it returns a Date? But that’s a little messy too, since that method requires an argument, and it’s unclear whether we should pass it a Date or a number… All D3 time intervals support coercion (as does Plot’s internal number interval), but should we expect users to implement coercion in their own intervals? I mean, isn’t the point of this PR is that they don’t? 🤔